common/src/main/java/com/wynntils/features/user/HealthPotionBlockerFeature.java

```diff
@@ -12,25 +13,45 @@
 import com.wynntils.wc.utils.WynnItemMatchers;
 import com.wynntils.wc.utils.WynnUtils;
 import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.network.protocol.game.ServerboundUseItemOnPacket;
 import net.minecraft.network.protocol.game.ServerboundUseItemPacket;
 import net.minecraft.world.item.ItemStack;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 @FeatureInfo(stability = Stability.STABLE)
 public class HealthPotionBlockerFeature extends UserFeature {
 
+    @Config
+    public static int threshold = 95;
+
     @SubscribeEvent
     public void onPotionUse(PacketSentEvent<ServerboundUseItemPacket> e) {
-        if (!WynnUtils.onWorld()) return;
+        Component response = getBlockResponse(e);
+        if (response != null) {
+            e.setCanceled(true);
+            McUtils.sendMessageToClient(response);
+        }
+    }
+
+    @SubscribeEvent
+    public void onPotionUseOn(PacketSentEvent<ServerboundUseItemOnPacket> e) {
+        if (getBlockResponse(e) != null) e.setCanceled(true);
```

Comment:
So we don't need to send a chat message here because the other event already sent one? Did you test this by looking at the sky for example?

Reply:
Yes, I checked looking at the sky too.